<launch>
    <arg name="pcd_path" default="$(find pointcloud_registration)/pcd_files" />
    <arg name="pointcloud_topic" default="/camera/depth_registered/points" />
    <arg name="write_file" default="false" />
    <arg name="rviz" default="true" />
    <arg name="fitness" default="0.6"/>
    <arg name="accumulation_seconds" value="1.0" />
    <arg name="voxel_size" value="0.005" />
    <arg name="registration_attempts" value="3" />

    <node name="pointcloud_registration_node" pkg="pointcloud_registration" type="registration_node.py" output="screen">
        <param name="pcd_path" value="$(arg pcd_path)" />
        <param name="pointcloud_topic" value="$(arg pointcloud_topic)" />
        <param name="write_file" value="$(arg write_file)" />
        <param name="fitness" value="$(arg fitness)" />
        <param name="accumulation_seconds" value="$(arg accumulation_seconds)" />
        <param name="voxel_size" value="$(arg voxel_size)" />
        <param name="registration_attempts" value="$(arg registration_attempts)" />
    </node>

    <!-- modify this -->
    <node pkg="tf" type="static_transform_publisher" name="map_to_camera" 
          args="0 0 0 0 0 0 map camera_link 100" />

    <group if="$(arg rviz)">
        <node pkg="rviz" type="rviz" name="display_rviz" 
            args="-d $(find pointcloud_registration)/rviz/display.rviz" />
    </group>

</launch>